packages/forms/src/Components/Select.php

```diff
@@ -1415,7 +1415,14 @@ public function getInValidationRuleValues(): ?array
             return array_keys($this->getOptionLabels(withDefaults: false));
         }
 
-        return blank($this->getOptionLabel(withDefault: false)) ? [] : null;
+        $optionLabel = $this->getOptionLabel(withDefault: false);
+
+        if ($this->isSearchable() && ! $optionLabel) {
```

Comment:
I don't think we can trust that an absence of the option label indicates an absence of `getOptionLabelUsing()`, it could be that the option is invalid and no label is returned. In which case, if its also searchable, this would mean the validation is skipped, I think.

To be honest, I think that we should probably alter `getOptionLabel()` and `getOptionLabels()` to throw an exception if `$withDefault` is `false` and `$this->getOptionLabelUsing` is null, instead of silently failing. Users should be forced to implement the correct callbacks otherwise validation is not possible.